internal/controller/clientpool/clientpool.go

```diff
@@ -44,11 +54,25 @@ func (cp *ClientPool) GetSDKClient(key ClientPoolKey) (sdkclient.Client, bool) {
 	cp.mux.RLock()
 	defer cp.mux.RUnlock()
 
-	c, ok := cp.clients[key]
-	if ok {
-		return c, true
+	info, ok := cp.clients[key]
+	if !ok {
+		return nil, false
 	}
-	return nil, false
+
+	// Check if any certificate is expired
+	if info.PEMCert != nil {
+		expired, err := cp.isCertificateExpiredSoon(info.PEMCert, 5*time.Minute)
```

Comment:
Can we avoid re-parsing the cert each time a client is fetched? This happens once for each reconciliation loop for each worker, so it could get expensive.